mobilitypath_visualizer/include/mobilitypath_visualizer.h

```diff
@@ -0,0 +1,125 @@
+#pragma once
+
+/*
+ * Copyright (C) 2020 LEIDOS.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+#include <ros/ros.h>
+#include <carma_utils/CARMAUtils.h>
+#include <visualization_msgs/MarkerArray.h>
+#include <string>
+#include <vector>
+#include <boost/uuid/uuid_generators.hpp>
+#include <boost/uuid/uuid_io.hpp>
+#include <tf2_ros/transform_listener.h>
+#include <tf2_geometry_msgs/tf2_geometry_msgs.h>
+#include <boost/shared_ptr.hpp>
+#include <cav_msgs/MobilityPath.h>
+#include <cav_msgs/BSM.h>
+#include <geometry_msgs/PoseStamped.h>
+#include <unordered_map>
+
+
+namespace mobilitypath_visualizer {
+
+    /**
+     * MobilityPathVisualizer TODO
+     * 
+    */ 
+    struct MarkerColor
+    {
+        double red = 0.0;
+        double green = 0.0;
+        double blue = 0.0;
+    };
+
+    class MobilityPathVisualizer
+    {
+        
+    public:
+
+        
+        /**
+         * \brief Default constructor
+         */
+        MobilityPathVisualizer();
+
+        /**
+         * \brief General starting point to run this node
+         */
+        void run();
+
+        visualization_msgs::MarkerArray composeVisualizationMarker(const cav_msgs::MobilityPath& msg, const tf2::Transform& map_in_earth, const MarkerColor& color);
+
+        visualization_msgs::MarkerArray composeLabelMarker(visualization_msgs::MarkerArray host_marker, std::vector<visualization_msgs::MarkerArray> cav_markers);
```

Comment:
Probably best to pass these inputs by const reference to keep it more efficient.